test/filters/test_crop_hull.cpp

```diff
@@ -0,0 +1,382 @@
+        cropHullFilter.setCropOutside(true);
+      }
+
+      pcl::PointCloud<pcl::PointXYZ> filteredCloud;
+      cropHullFilter.filter(filteredCloud);
+      ASSERT_EQ (expectedCloud.size(), filteredCloud.size());
+      for (pcl::index_t i = 0; i < expectedCloud.size(); ++i)
+      {
+        EXPECT_XYZ_NEAR(expectedCloud[i], filteredCloud[i], 1e-5);
+      }
+      // check non empty out cloud filtering
+      cropHullFilter.filter(filteredCloud);
+      EXPECT_EQ (expectedCloud.size(), filteredCloud.size());
+
+      // check keep organized
+      {
+        cropHullFilter.setKeepOrganized(true);
+        cropHullFilter.setUserFilterValue(-10.);
+        pcl::PointXYZ defaultPoint(-10., -10., -10.);
+        cropHullFilter.filter(filteredCloud);
+        ASSERT_EQ (inputCloud->size(), filteredCloud.size());
+        for (pcl::index_t i = 0; i < inputCloud->size(); ++i)
+        {
+          pcl::PointXYZ expectedPoint = expectedFilteredMask[i] ? inputCloud->at(i) : defaultPoint;
+          EXPECT_XYZ_NEAR(expectedPoint, filteredCloud[i], 1e-5);
+        }
+        cropHullFilter.setKeepOrganized(false);
+      }
+    }
+  };
+
+  bool randomBool() {
+    static auto gen = std::bind(
+        std::uniform_int_distribution<>(0,1),
+        std::default_random_engine());
+    return gen();
+  }
+
+  std::vector<std::pair<pcl::Indices, pcl::PointCloud<pcl::PointXYZ>::ConstPtr>>
+  createTestData(
+      std::function<pcl::PointXYZ()> insidePointGenerator,
+      std::function<pcl::PointXYZ()> outsidePointGenerator)
+  {
+    std::vector<std::pair<pcl::Indices, pcl::PointCloud<pcl::PointXYZ>::ConstPtr>> testData;
+    size_t const chunkSize = 1000;
+    pcl::PointCloud<pcl::PointXYZ>::Ptr insideCloud(new pcl::PointCloud<pcl::PointXYZ>);
+    pcl::PointCloud<pcl::PointXYZ>::Ptr outsideCloud(new pcl::PointCloud<pcl::PointXYZ>);
+    pcl::PointCloud<pcl::PointXYZ>::Ptr mixedCloud(new pcl::PointCloud<pcl::PointXYZ>);
+    pcl::Indices insideIndicesForInsideCloud;
+    pcl::Indices insideIndicesForOutsideCloud;
+    pcl::Indices insideIndicesForMixedCloud;
+    for (size_t i = 0; i < chunkSize; ++i)
+    {
+      insideIndicesForInsideCloud.push_back(i);
+      insideCloud->push_back(insidePointGenerator());
+      outsideCloud->push_back(outsidePointGenerator());
+      if (randomBool()) {
+        insideIndicesForMixedCloud.push_back(i);
+        mixedCloud->push_back(insidePointGenerator());
+      }
+      else {
+        mixedCloud->push_back(outsidePointGenerator());
+      }
+    }
+    testData.emplace_back(insideIndicesForInsideCloud, insideCloud);
+    testData.emplace_back(insideIndicesForOutsideCloud, outsideCloud);
+    testData.emplace_back(insideIndicesForMixedCloud, mixedCloud);
+    return testData;
+  }
+}
+
+
+//////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+TEST (PCL, ConvexHull_2dsquare)
+{
```

Comment:
This is actually 2 tests. One L182-L205 and another from L207-226. Each with a different setup (as explained in (2))